test/iaas.AzureClient.spec.js

```diff
@@ -0,0 +1,366 @@
+      const blobName = 'blob1';
+
+      afterEach(function () {
+        mocks.reset();
+      });
+
+      it('blob list operation should be successful', function () {
+        const blobList = ['blob1.txt', 'blob2.txt'];
+        const expectedResponse = {
+          headers: {
+            'transfer-encoding': 'chunked',
+            'content-type': 'application/xml',
+            server: 'Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0',
+            'x-ms-request-id': '2376cafa-0001-002e-1e01-6716c0000000',
+            'x-ms-version': '2016-05-31',
+            date: new Date().toISOString(),
+            connection: 'close'
+          },
+          status: 200,
+          body: `<?xml version="1.0" encoding="utf-8"?>  
+          <EnumerationResults ContainerName="https://${settings.storageAccount}.blob.core.windows.net/${settings.container}">  
+            <Blobs>
+              <Blob>
+                <Name>${blobList[0]}</Name>  
+                <Url>https://${settings.storageAccount}.blob.core.windows.net/${settings.container}/${blobList[0]}</Url>  
+                <Properties>
+                  <Last-Modified>Wed, 09 Sep 2016 09:20:02 GMT</Last-Modified>  
+                  <Etag>0x8CBFF45D8A29A19</Etag>  
+                  <Content-Length>100</Content-Length>  
+                  <Content-Type>text/html</Content-Type>  
+                  <Content-Encoding />  
+                  <Content-Language>en-US</Content-Language>  
+                  <Content-MD5 />  
+                  <Cache-Control>no-cache</Cache-Control>  
+                  <BlobType>BlockBlob</BlobType>  
+                  <LeaseStatus>unlocked</LeaseStatus>  
+                </Properties>
+                <Metadata>  
+                  <Color>blue</Color>  
+                  <BlobNumber>01</BlobNumber>  
+                </Metadata>  
+              </Blob>
+              <Blob>
+                <Name>${blobList[1]}</Name>
+                <Url>https://${settings.storageAccount}.blob.core.windows.net/${settings.container}/${blobList[1]}</Url>  
+                <Properties>  
+                  <Last-Modified>Wed, 09 Sep 2016 09:20:02 GMT</Last-Modified>  
+                  <Etag>0x8CBFF45D8B4C212</Etag>  
+                  <Content-Length>5000</Content-Length>  
+                  <Content-Type>application/octet-stream</Content-Type>  
+                  <Content-Encoding>gzip</Content-Encoding>  
+                  <Content-Language />  
+                  <Content-MD5 />  
+                  <Cache-Control />  
+                  <BlobType>BlockBlob</BlobType>  
+                </Properties>  
+                <Metadata>  
+                  <Color>green</Color>  
+                  <BlobNumber>02</BlobNumber>  
+                </Metadata>  
+              </Blob>
+            </Blobs>  
+            <NextMarker />   
+          </EnumerationResults>`
+        };
+        mocks.azureClient.list(settings.container, undefined, expectedResponse);
```

Comment:
Is xml the expected response type? most APIs now have support for JSON can you check if you can tweak any input configs for it to directly return JSON instead of XML. [XML parsing is slower than compared to JSON parsing]

Reply:
Azure supports XML as of now. See https://docs.microsoft.com/en-us/rest/api/storageservices/list-blobs. I didn't find json example response anywhere.

Reply:
ok